src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ConsoleDebugTargetsProvider.cs

```diff
@@ -0,0 +1,322 @@
+        {
+            if(string.IsNullOrEmpty(executable))
+            {
+                throw new Exception(string.Format(VSResources.NoDebugExecutableSpecified, profileName));
+            }
+            else if(executable.IndexOf(Path.DirectorySeparatorChar) != -1 && !TheFileSystem.FileExists(executable))
+            {
+                throw new Exception(string.Format(VSResources.DebugExecutableNotFound, executable, profileName));
+            }
+            else if(!string.IsNullOrEmpty(workingDir) && !TheFileSystem.DirectoryExists(workingDir))
+            {
+                throw new Exception(string.Format(VSResources.WorkingDirecotryInvalid, workingDir, profileName));
+            }
+        }
+
+        /// <summary>
+        /// Helper returns cmd.exe as the launcher for Ctrl-F5 (useCmdShell == true), otherwise just the exe and args passed in.
+        /// </summary>
+        public void GetExeAndArguments(bool useCmdShell, string debugExe, string debugArgs, out string finalExePath, out string finalArguments)
+        {
+            if(useCmdShell)
+            {
+                // Escape the characters ^<>& so that they are passed to the application rather than interpreted by cmd.exe.
+                string escapedArgs = string.Empty;
+                if(!string.IsNullOrWhiteSpace(debugArgs))
+                {
+                    escapedArgs = debugArgs.Replace("^","^^");
+                    escapedArgs = escapedArgs.Replace("<","^<");
+                    escapedArgs = escapedArgs.Replace(">","^>");
+                    escapedArgs = escapedArgs.Replace("&","^&");
+                }
+                finalArguments = string.Format("/c \"\"{0}\" {1} & pause\"", debugExe, escapedArgs);
+                finalExePath = Path.Combine(Environment.SystemDirectory, "cmd.exe");
+            }
+            else
+            {
+                finalArguments = debugArgs;
+                finalExePath = debugExe;
+            }
+        }
+    
+        /// <summary>
+        /// This is called on F5 to return the list of debug targets. What we return depends on the type
+        /// of project.  
+        /// </summary>
+        private async Task<DebugLaunchSettings> GetConsoleTargetForProfile(ILaunchProfile resolvedProfile, DebugLaunchOptions launchOptions, 
+                                                              string projectFolder, bool useCmdShell)
+        {
+            var settings = new DebugLaunchSettings(launchOptions);
+
+            string executable, arguments;
+            string commandLineArguments = resolvedProfile.CommandLineArgs;
+
+            // Is this profile just running the project? If so we ignore the exe
+            if(string.Equals(resolvedProfile.CommandName, LaunchSettingsProvider.RunProjectCommandName, StringComparison.OrdinalIgnoreCase))
+            {
+                // Can't run a class library directly
+                if(await GetIsClassLibraryAsync().ConfigureAwait(false))
```

Comment:
I remember @srivatsn mentioning that a class library can be executed by setting an entry execution point(I dont know how). I do see the comment which mentions `directly`. But I still wanted to make sure if this is the intented behavior